implementation/contracts/deposit/DepositFunding.sol

```diff
@@ -65,6 +65,7 @@ library DepositFunding {
         uint256 _bondRequirementSatoshi = _lotSizeSatoshis.mul(_system.getInitialCollateralizedPercent()).div(100);
         uint256 _bondRequirementWei = _d.fetchBitcoinPrice().mul(_bondRequirementSatoshi);
 
+        _d.keepSetupFee = _system.createNewDepositFeeEstimate();
```

Comment:
This feels like it should be `= msg.value`, no? That's what we actually forward to the keep, should be what we capture here also.

Reply:
Thought so initially, but `msg.value` is unbounded and that's a potential hole. It's not very realistic so if you think it's not worth covering, we record msg.value

Reply:
I see… So the depositor could send a greater `msg.value` than the bond amount, and then when we try and refund them we have insufficient seized bond to do it, and things go sideways. We also give the depositor the ability to take more from signers than they are really due in the case where signer setup fails.

I think the thing we need to do here is use the lesser of `msg.sender` and `_system.createNewDepositFeeEstimate`. This protects against situations where the depositor sends more or less than the deposit fee estimate and then gets repaid. Thoughts?

Reply:
if `createNewDepositFeeEstimate ` is less than `msg.value` does the tx still go through? I'm not sure how this works on the Keep side

Reply:
I believe it goes through + surplus is captured in the system to subsidize future entries.

Reply:
Let's chat about this tomorrow, I think. It feels like just a minor griefing vector, since the depositor can only regain as much as they sent… It only works to specifically target signers and take bond from them, not as a way to gain any value.

Reply:
> surplus is captured in the system to subsidize future entries.

My q was with msg.value < createNewDepositFeeEstimate case, Don't know if this fails on Keep side. 

> Let's chat about this tomorrow, I think. It feels like just a minor griefing vector

Agree, I's have a bigger issue with it if the Depositor has a way to intentionally trigger it. I would still like to avoid it though

Reply:
> My q was with msg.value < createNewDepositFeeEstimate case, Don't know if this fails on Keep side.

`msg.value < openKeepFeeEstimate()` will revert on `openKeep` call.

Reply:
Okay, let's aim to capture `msg.value` and use that for refunding.

Reply:
What is the rationale for `msg.value` over `openKeepFeeEstimate`?
It protects users from receiving less if they overfunded, but I don't think that should be at the signers' expense

Reply:
Yeah I don't see a great way out of this.

If we refund based on the estimate, and the estimate is less than the provided value, the funder loses the spread without doing anything wrong. If everyone is behaving correctly and honestly, the value loss is small. If someone is misbehaving, the value loss is still bounded and small, since the keep estimates are part of the system rather than controlled by any of the parties in this particular interaction.

If we refund based on the value, the funder gets fully refunded, but may take more than they're really due from the signers (though notably, the signers have failed to do what they were supposed to at this point). Assuming everyone is behaving correctly and honestly, the value loss would once again be small. If someone is misbehaving, there is potentially high value loss (up to the full signer bond). The misbehavior required is sizeable, in that the funder has to also be able to prevent the signing group from forming; however, if the lot size is high enough and the bonds are correspondingly large enough, this may be enticing.

If we do the lesser of the two, we're really not getting any improvement over always refunding based on the current estimate. If we do the greater of the two, then we're not getting any improvement over refunding the funder value. So those two options aren't really worth it.

Given this, I think staying with the current fee estimate feels like the right next move. I still want to talk about this tomorrow (which we didn't do last time I said it hah) -- @NicholasDotSol do you mind making sure we have this conversation?